tests/foundry/src/IBC.t.sol

```diff
@@ -0,0 +1,192 @@
+    }
+
+    function setUpMockClient() internal {
+        handler.createClient(IBCMsgs.MsgCreateClient({
+            clientType: mockClientType,
+            height: Height.Data({revision_number: 0, revision_height: 1}),
+            clientStateBytes: wrapMockClientState(IbcLightclientsMockV1ClientState.Data({latest_height: Height.Data({revision_number: 0, revision_height: 1})})),
+            consensusStateBytes: wrapMockConsensusState(IbcLightclientsMockV1ConsensusState.Data({timestamp: uint64(block.timestamp)}))
+        }));
+    }
+
+    function setUpConnection() internal {
+        address prev = host.getIBCModule();
+        host.setIBCModule(address(this));
+        ConnectionEnd.Data memory connection = ConnectionEnd.Data({
+            client_id: "mock-client-0",
+            versions: getConnectionVersions(),
+            state: ConnectionEnd.State.STATE_OPEN,
+            delay_period: 0,
+            counterparty: Counterparty.Data({
+                client_id: "mock-client-0",
+                connection_id: "connection-0",
+                prefix: MerklePrefix.Data({key_prefix: bytes("ibc")})
+            })
+        });
+        host.setConnection("connection-0", connection);
+        host.setIBCModule(prev);
+    }
+
+    function setUpChannel() internal {
+        address prev = host.getIBCModule();
+        host.setIBCModule(address(this));
+        string[] memory hops = new string[](1);
+        hops[0] = "connection-0";
+        Channel.Data memory channel = Channel.Data({
+            state: Channel.State.STATE_OPEN,
+            ordering: Channel.Order.ORDER_UNORDERED,
+            counterparty: ChannelCounterparty.Data({
+                port_id: portId,
+                channel_id: "channel-0"
+            }),
+            connection_hops: hops,
+            version: "1"
+        });
+        host.setChannel(portId, "channel-0", channel);
+        host.setNextSequenceSend(portId, "channel-0", 1);
+        host.setNextSequenceRecv(portId, "channel-0", 1);
+        host.setNextSequenceAck(portId, "channel-0", 1);
+        host.setIBCModule(prev);
+
+        testPacketCommitment = makePacketCommitment(getPacket());
+    }
+
+    function setUpMockApp() internal {
+        mockApp = new MockApp();
+        handler.bindPort(portId, address(mockApp));
+
+        address prev = host.getIBCModule();
+        host.setIBCModule(address(this));
+        host.claimCapability(IBCIdentifier.channelCapabilityPath(portId, "channel-0"), address(mockApp));
+        host.claimCapability(IBCIdentifier.channelCapabilityPath(portId, "channel-0"), address(this));
+        host.setIBCModule(prev);
+    }
+
+    /* test cases */
+
+    function testClientRegistration() public view {
+        (address client, bool found) = host.getClientImpl(mockClientType);
+        assert(found);
+        assert(address(mockClient) == client);
+    }
+
+    function testSendPacket() public {
+        Packet.Data memory packet = Packet.Data({
```

Comment:
Why don't you use `getPacket()` to make an instance of `Packet.Data`?

Reply:
fixed at https://github.com/hyperledger-labs/yui-ibc-solidity/pull/92/commits/d8c164a8a2862fd4732a45d744a12b30477287a7